test/src/unit-filter-pipeline.cc

```diff
@@ -4048,4 +4049,87 @@ TEMPLATE_TEST_CASE(
     int64_t) {
   testing_float_scaling_filter<float, TestType>();
   testing_float_scaling_filter<double, TestType>();
-}
\ No newline at end of file
+}
+
+template <typename T>
+void testing_xor_filter() {
+  tiledb::sm::Config config;
+
+  // Set up test data
+  const uint64_t nelts = 100;
+  const uint64_t tile_size = nelts * sizeof(T);
+  const uint64_t cell_size = sizeof(T);
+  const uint32_t dim_num = 0;
+
+  Tile tile;
+  Datatype t = Datatype::INT8;
+  switch (sizeof(T)) {
+    case 1: {
+      t = Datatype::INT8;
+    } break;
+    case 2: {
+      t = Datatype::INT16;
+    } break;
+    case 4: {
+      t = Datatype::INT32;
+    } break;
+    case 8: {
+      t = Datatype::INT64;
+    } break;
+    default: {
+      INFO(
+          "testing_xor_filter: passed int type with size of not 1, 2, 4, or 8 "
+          "bytes.");
+      CHECK(false);
+    }
+  }
+
+  tile.init_unfiltered(
+      constants::format_version, t, tile_size, cell_size, dim_num);
+
+  // The MSVC Windows compiler doesn't support std::uniform_int_distribution
+  // with
+  // type int8_t instantiation, so we use this little hack to generate the right
+  // type for our int distribution.
+  std::random_device rd;
```

Comment:
Actually random seeds are bad for unit testing because they're not repeatable. A fixed seed is required here.